src/Services/BuddyService.cs

```diff
@@ -25,20 +25,19 @@ public BuddyService(DBContext ctx, IOptions<ApiServerConfig> config, MessageServ
         public BuddyList GetBuddyList(Viking viking)
         {
             List<Schema.Buddy> buddies = new List<Schema.Buddy>();
-            foreach(Model.Buddy buddy in viking.Buddies)
-            {
-                Model.Buddy otherBuddy = ctx.Buddies.FirstOrDefault(e => e.OwnerID == buddy.BuddyID);
 
-                if (otherBuddy == null) continue;
+            List<Model.Buddy> vikingBuddies = ctx.Buddies.Where(e => e.OwnerID == viking.Id).ToList(); // relations aren't right, getting around it
```

Comment:
so why don't fix relation instead of placing this workaround in the next places?